File: exercises/Week/2/JS/OOP/class_person.js
class person{
    fName;
    lName;
    age;
    gender;


    constructor(fName, lName, age, gender){
        this.fName = fName;
        this.lName = lName;
        this.age = age;
        this.gender = gender;
    }

    introduce(){
        let msg = "Hi I'm " + this.fName + " " + this.lName;
        console.log(msg)
    }

    getInfo(){
        let msg = "First name: " + this.fName + "\n" +
                  "Last name: " + this.lName + "\n" +
                  "Age: " + this.age + "\n" +
                  "Gender: " + this.gender + "\n";
                  
        console.log(msg)
    }
}

let p1 = new person("a", "b", 10, "male");
p1.introduce();

let msg = "First name: " + p1.fName;
console.log(msg)
msg = "Last name: " + p1.lName;
console.log(msg);
msg = "Age: " + p1.age;
console.log(msg);
msg = "Gender: " + p1.gender;
console.log(msg);

let p2 = new person("b", "c", 10, "female");
p2.introduce();
p2.getInfo();
p1.getInfo();
p2.getInfo();

File: exercises/Week/2/JS/OOP/class_person.cs
class person {
    public string fName;
    public string lName;
    public string gender;
    public int age;

/*
    person (fName, lName) {
        this.fName = fName;
        this.lName = lName;
        this.age = 0;
        this.gender = "";
    }

    person (fname, lName, age){
        this.fName = fName;
        this.lName = lName;
        this.age = age;
        this.gender = "";

    }
*/
    person (fName, lName, gender, age){
        this.fName = fName;
        this.lName = lName;
        this.age = age;
        this.gender = gender;
    }
   public string getInfo {
        string msg =    "First name: " + this.fName + "\n" +
                        "Last name: " + this.lName + "\n" +
                        "Age: " + age + "\n" +
                        "Gender: " + gender + "\n";
        return msg;
    }

}

//static void main (string[] args) {
    /*
    person p1 = new person("a", "b", 18, "female");
    string info;
    info = p1.getInfo();
    */
    Console.WriteLine("test");
